modules/shBidAdapter.js

```diff
@@ -154,25 +169,95 @@ function createBids(bidRes, reqData) {
     }
     if (reqBid.mediaType === BANNER) {
       bidUnit.ad = getBannerHtml(bid, reqBid, reqData);
+    } else if (reqBid.context === 'outstream') {
+      const renderer = Renderer.install({
+        id: bid.bidId,
+        url: '//',
+        config: {
+          playerId: reqBid.playerId,
+          width: bid.video.width,
+          height: bid.video.height,
+          vastUrl: bid.vastTag,
+          vastXml: bid.vastXml,
+          debug: reqData.debug,
+          isStage: !!reqData.meta.stage,
+          customRender: utils.getBidIdParameter('customRender', currentBidRequest.params.outstreamOptions),
+          slot: utils.getBidIdParameter('slot', currentBidRequest.params.outstreamOptions),
+          iframe: utils.getBidIdParameter('iframe', currentBidRequest.params.outstreamOptions),
+        }
+      });
+      renderer.setRender(outstreamRender);
+      bidUnit.renderer = renderer;
     }
     bids.push(bidUnit);
   });
 
   return bids;
 }
 
+function outstreamRender(bid) {
+  const embedCode = createOutstreamEmbedCode(bid);
+  if (typeof bid.renderer.config.customRender === 'function') {
+    bid.renderer.config.customRender(bid, embedCode);
+  } else {
+    try {
+      const inIframe = utils.getBidIdParameter('iframe', bid.renderer.config);
+      if (inIframe && window.document.getElementById(inIframe).nodeName === 'IFRAME') {
+        const iframe = window.document.getElementById(inIframe);
+        let framedoc = iframe.contentDocument || (iframe.contentWindow && iframe.contentWindow.document);
+        framedoc.body.appendChild(embedCode);
+        return;
+      }
+
+      const slot = utils.getBidIdParameter('slot', bid.renderer.config);
+      if (slot && window.document.getElementById(slot)) {
+        window.document.getElementById(slot).appendChild(embedCode);
+      } else if (slot) {
+        utils.logError('[ShowHeroes][renderer] Error: spot not found');
+      }
```

Comment:
```suggestion
      }
      // Served in an iframe by an ad server
      window.document.body.appendChild(embedCode);
```

The last case would be to just insert the `embedCode` inside the iframe.